bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/validation/IpAddressValidator.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.haassohnpelletoven.validation;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link IpAddressValidator} is responsible for validating a given IP-Address.
+ *
+ * @author Christian Feininger - Initial contribution
+ */
+@NonNullByDefault
+public class IpAddressValidator {
+
+    private static final String zeroTo255 = "([01]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])";
+
+    private static final String IP_REGEXP = zeroTo255 + "\\." + zeroTo255 + "\\." + zeroTo255 + "\\." + zeroTo255;
+
+    private static final Pattern IP_PATTERN = Pattern.compile(IP_REGEXP);
```

Comment:
Actually the framework takes care that a valid IP address or hostname is specified. Is there a reason why you make an additional validation here?

Reply:
no, I didn't know that. I would like to keep it anyway to assure a right error messaging and it is working though.

Reply:
The UI will produce a correct error message. So, your code would be redundant. For the sake of better code maintainability, it's better to remove it.

Reply:
ok